src/scheduler/job_info.c

```diff
@@ -3426,11 +3429,42 @@ find_jobs_to_preempt(status *policy, resource_resv *hjob, server_info *sinfo, in
 
 		update_universe_on_end(npolicy, pjob,  "S", NO_ALLPART);
 		rjobs_count--;
+		/* Check if any of the previously preempted job increased its preemption priority */
+		for (ind = 0; pjobs[ind] != NULL; ind++) {
+			if (pjobs[ind]->job->preempt >= nhjob->job->preempt) {
+				dont_preempt_job = 1;
+				break;
+			}
+		}
+		/* Check if the job we just ended increases its priority. If so, don't preempt this job */
```

Comment:
So, this if statement isn't really checking that pjob's p-priority increased, it's checking whether its p-priority increased more than high priority job's, right? If so, can you please update the comment to say that instead?